docs/yul.rst

```diff
@@ -559,21 +566,15 @@ syntactical scoping rules.
 This means a for-loop of the form ``for { I... } C { P... } { B... }`` is equivalent
 to ``{ I... for {} C { P... } { B... } }``.
 
-
 The parameters and return parameters of functions are visible in the
 function body and their names have to be distinct.
 
-Variables can only be referenced after their declaration. In particular,
-variables cannot be referenced in the right hand side of their own variable
-declaration.
-Functions can be referenced already before their declaration (if they are visible).
+Inside functions, it is not possible to reference a variable that was declared
+outside of that function.
 
```

Comment:
```suggestion
```

Reply:
This is what I meant in https://github.com/ethereum/solidity/pull/11726/files#r680945865

Reply:
Hm, I think shadowing is important enough so that it gets its own paragraph. And the one before is still about functions and visibilty.

Reply:
Isn't shadowing only relevant for functions?

Reply:
No, also inside blocks:
```
{
let x
  {
    let x
  }
}
```